lib/rpc/request.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Soroban.RPC.Request do
+  @moduledoc """
+  A module to work with Soroban RPC requests.
+  Requests are composed in a functional manner.
+  The request does not happen until it is configured and passed to `perform/1`.
+  """
+
+  alias Soroban.RPC.{Client, Error, HTTPError}
+
+  @default_url "https://rpc-futurenet.stellar.org:443/"
+
+  @type endpoint :: String.t()
+  @type headers :: [{binary(), binary()}]
+  @type params :: map() | nil
+  @type opts :: Keyword.t()
+  @type response :: {:ok, map()} | {:error, Error.t()} | {:error, HTTPError.t()}
+  @type parsed_response :: {:ok, struct()} | {:error, Error.t()} | {:error, HTTPError.t()}
+  @type url :: String.t()
+
+  @type t :: %__MODULE__{
+          endpoint: endpoint(),
+          url: url(),
+          params: params(),
+          headers: headers()
+        }
+
+  defstruct [
+    :endpoint,
+    :url,
+    :params,
+    :headers
+  ]
+
+  @spec new(endpoint :: endpoint(), opts :: opts()) :: t()
+  def new(endpoint, opts \\ []) do
+    url = Keyword.get(opts, :url, @default_url)
+
+    %__MODULE__{
+      endpoint: endpoint,
+      url: url,
+      params: nil,
+      headers: []
+    }
+  end
+
+  @spec add_headers(request :: t(), headers :: headers()) :: t()
+  def add_headers(%__MODULE__{} = request, headers), do: %{request | headers: headers}
+
+  @spec add_params(request :: t(), params :: params()) :: t()
+  def add_params(%__MODULE__{} = request, params),
+    do: %{request | params: params}
+
+  @spec perform(request :: t()) :: response()
+  def perform(%__MODULE__{endpoint: endpoint, url: url, headers: headers, params: params}) do
+    with {:ok, %{result: result}} <- Client.request(endpoint, url, headers, params) do
+      {:ok, result}
+    end
+  end
```

Comment:
IMO we could avoid the `with` statement by returning directly the result from `client/default.ex`

And here we only call: `Client.request(endpoint, url, headers, params)`


What do you think?

Reply:
But we need to return only the result from the response `{:ok, %{result: result}}`

Reply:
Yeah, I mea, we could change this to:
https://github.com/kommitters/soroban.ex/blob/v0.3/lib/rpc/client/default.ex#L45-L48

```elixir
    case decoded_body do
      %{error: error} -> {:error, Error.new(error)}
      %{result: result} -> {:ok, result}
    end
```

Does it work?

Reply:
Sounds good!